vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/MongoClientImpl.java

```diff
@@ -319,6 +317,59 @@ public void close() {
     return this;
   }
 
+  @Override
+  public io.vertx.ext.mongo.MongoClient distinct(String collection, String fieldName, String resultClassname, Handler<AsyncResult<JsonArray>> resultHandler) {
+    requireNonNull(collection, "collection cannot be null");
+    requireNonNull(fieldName, "fieldName cannot be null");
+    requireNonNull(resultHandler, "resultHandler cannot be null");
+
+    MongoCollection<JsonObject> coll = getCollection(collection);
+    try {
+      DistinctIterable distinctValues = coll.distinct(fieldName, Class.forName(resultClassname));
+      List results = new ArrayList();
+      distinctValues.into(results, (result, error) -> {
+        vertx.runOnContext(v -> {
+          if (error != null) {
```

Comment:
May be 'v' is an acronym for 'veryshortnameforvoidaction' or a tribute to an old 80's serie ;-)  
btw I guess it was a convention to name this variable "v" because all  runOnContext in vert-core use "v" name.